src/Symfony/Component/Serializer/Normalizer/SerializerAwareNormalizer.php

```diff
@@ -11,26 +11,17 @@
 
 namespace Symfony\Component\Serializer\Normalizer;
 
-use Symfony\Component\Serializer\SerializerInterface;
+use Symfony\Component\Serializer\SerializerAwareTrait;
 use Symfony\Component\Serializer\SerializerAwareInterface;
 
 /**
  * SerializerAware Normalizer implementation.
  *
  * @author Jordi Boggiano <j.boggiano@seld.be>
+ *
+ * @deprecated since version 3.1, to be removed in 4.0. Use the SerializerAwareTrait instead.
```

Comment:
You must trigger a deprecation error like this one: `@trigger_error(sprintf('The class "%s" is deprecated. It will be removed in Symfony 4.0. Use the "Symfony\Component\Serializer\SerializerAwareTrait" trait instead.', __CLASS__), E_USER_DEPRECATED);`

You should also adapt existing normalizers in Symfony to avoid using this deprecated class. But if you do that... It will be a small BC break (`ObjectNormalizer` will not inherit from the `SerializerAwareNormalizer` type anymore - but as there is a corresponding interface, the BC break should be very limited). What do you think @symfony/deciders?